monitoring/collectors/jenkins_collector_adapter.py

```diff
@@ -0,0 +1,58 @@
+import requests
+from stats import BuildStats, StageStats
+
+
+class InvalidResponseCodeError(Exception):
+    pass
+
+
+class JenkinsCollectorAdapter(object):
+
+    def __init__(self, job_name, build_url):
+        self.job_name = job_name
+        self.url = build_url
+
+
+    def collect(self):
+        resp = self._get_raw_stats_from_jenkins()
+        return self._convert_raw_stats_to_build_stats(resp)
+
+
+    def _get_raw_stats_from_jenkins(self):
+        endpoint = self._get_build_stats_endpoint(self.url)
+        resp = requests.get(endpoint)
+
+        if resp.status_code != 200:
+            raise InvalidResponseCodeError()
+
+        return resp.json()
+
+
+    def _convert_raw_stats_to_build_stats(self, raw_stats):
+        timestamp = int(raw_stats['endTimeMillis'] / 1000)
+
+        stages_stats = []
+
+        if 'stages' in raw_stats.keys():
+            stages_stats = [StageStats(
+                name = raw_stage['name'],
+                status = raw_stage['status'],
+                duration_millis = raw_stage['durationMillis']
+            ) for raw_stage in raw_stats['stages']]
```

Comment:
This is pretty minimal and pedantic, but I confused raw_stage with raw_stats and I lost 5 seconds of my life figuring it out. Maybe rename raw_stage to x or stage or something which doesn't have levenshtein distance of 2.